dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -164,3 +207,142 @@ def dpnp_eigh(a, UPLO):
         ht_copy_ev.wait()
 
         return w, out_v
+
+
+def dpnp_solve(a, b):
+    """
+    dpnp_solve(a, b)
+
+    Return the solution to the system of linear equations with
+    a square coefficient matrix `a` and multiple dependent variables
+    array `b`.
+
+    """
+
+    a_usm_arr = dpnp.get_usm_ndarray(a)
+    b_usm_arr = dpnp.get_usm_ndarray(b)
+
+    b_order = "C" if b.flags.c_contiguous else "F"
+    a_shape = a.shape
+    b_shape = b.shape
+
+    exec_q = dpctl.utils.get_execution_queue((a.sycl_queue, b.sycl_queue))
+    if exec_q is None:
+        raise ValueError(
+            "Execution placement can not be unambiguously inferred "
+            "from input arguments."
+        )
+
+    res_type = _common_type(a, b)
+    if b.size == 0:
+        return dpnp.empty(b_shape, dtype=res_type)
```

Comment:
this doesn't follow CFD, since the allocation is made with default queue, so you need something like:
```suggestion
        return dpnp.empty_like(b, dtype=res_type)
```